stdlib/public/Concurrency/TaskCancellation.swift

```diff
@@ -0,0 +1,171 @@
+////===----------------------------------------------------------------------===//
+////
+//// This source file is part of the Swift.org open source project
+////
+//// Copyright (c) 2020 Apple Inc. and the Swift project authors
+//// Licensed under Apache License v2.0 with Runtime Library Exception
+////
+//// See https://swift.org/LICENSE.txt for license information
+//// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+////
+////===----------------------------------------------------------------------===//
+
+import Swift
+import Dispatch
+@_implementationOnly import _SwiftConcurrencyShims
+
+// ==== Task Cancellation ------------------------------------------------------
+
+extension Task {
+
+  /// Returns `true` if the task is cancelled, and should stop executing.
+  ///
+  /// - SeeAlso: `Task.checkCancellation`
+  public static func isCancelled() async -> Bool {
+    // pseudo code:
+    // let task = Task.current
+    // if task.cancelled { // using context metadata storage
+    //   return true
+    // } else if let deadline = task.deadline {
+    //   return deadline < DispatchWallTime.now()
```

Comment:
Should this be `deadline > DisprchWallTime.now` / `deadline.isOverdue`?

Reply:
yup, should use `isOverdue`, code in comments eh :) Thanks though 👍